provider/marathon/marathon_test.go

```diff
@@ -641,184 +517,82 @@ func TestMarathonGetPort(t *testing.T) {
 		expected    string
 	}{
 		{
-			desc: "port missing",
-			application: marathon.Application{
-				ID:     "app",
-				Labels: &map[string]string{},
-			},
-			task: marathon.Task{
-				AppID: "app",
-				Ports: []int{},
-			},
-			expected: "",
+			desc:        "port missing",
+			application: application(),
+			task:        task(),
+			expected:    "",
 		},
 		{
-			desc: "explicit port taken",
-			application: marathon.Application{
-				ID: "app",
-				Labels: &map[string]string{
-					types.LabelPort: "80",
-				},
-			},
-			task: marathon.Task{
-				AppID: "app",
-				Ports: []int{},
-			},
+			desc: "numeric port",
+			application: application(labels(map[string]string{
+				types.LabelPort: "80",
+			})),
+			task:     task(),
 			expected: "80",
 		},
 		{
-			desc: "illegal explicit port specified",
-			application: marathon.Application{
-				ID: "app",
-				Labels: &map[string]string{
-					types.LabelPort: "foobar",
-				},
-			},
-			task: marathon.Task{
-				AppID: "app",
-				Ports: []int{80},
-			},
+			desc: "string port",
+			application: application(labels(map[string]string{
+				types.LabelPort: "foobar",
+			})),
+			task:     task(taskPorts(80)),
 			expected: "",
 		},
 		{
-			desc: "illegal explicit port integer specified",
-			application: marathon.Application{
-				ID: "app",
-				Labels: &map[string]string{
-					types.LabelPort: "-1",
-				},
-			},
-			task: marathon.Task{
-				AppID: "app",
-				Ports: []int{80},
-			},
+			desc: "negative port",
+			application: application(labels(map[string]string{
+				types.LabelPort: "-1",
+			})),
+			task:     task(taskPorts(80)),
 			expected: "",
 		},
 		{
-			desc: "task port available",
-			application: marathon.Application{
-				ID:     "app",
-				Labels: &map[string]string{},
-				PortDefinitions: &[]marathon.PortDefinition{
-					{
-						Port: testhelpers.Intp(443),
-					},
-				},
-				IPAddressPerTask: &marathon.IPAddressPerTask{
-					Discovery: &marathon.Discovery{
-						Ports: &[]marathon.Port{
-							{
-								Number: 8000,
-							},
-						},
-					},
-				},
-			},
-			task: marathon.Task{
-				AppID: "app",
-				Ports: []int{80},
-			},
-			expected: "80",
+			desc:        "task port available",
+			application: application(),
+			task:        task(taskPorts(80)),
+			expected:    "80",
 		},
 		{
-			desc: "port mapping port available",
-			application: marathon.Application{
-				ID:     "app",
-				Labels: &map[string]string{},
-				PortDefinitions: &[]marathon.PortDefinition{
-					{
-						Port: testhelpers.Intp(443),
-					},
-				},
-				IPAddressPerTask: &marathon.IPAddressPerTask{
-					Discovery: &marathon.Discovery{
-						Ports: &[]marathon.Port{
-							{
-								Number: 8000,
-							},
-						},
-					},
-				},
-			},
-			task: marathon.Task{
-				AppID: "app",
-				Ports: []int{},
-			},
+			desc: "port definition available",
+			application: application(
```

Comment:
I did remove it intentionally for two reasons (should have commented this in the first place, sorry):

1. My original thinking was that the IP-per-task setup would serve as a safeguard to running into an undesirable happy path due to a bug in the code (i.e., false positive). On closer inspection, this seems highly unlikely since we can't really produce the expected port result of 443 without configuring it somewhere, which we only do for the test case at hand that we intend to cover.
1. Configuring both port definitions and IPs-per-task should actually not be allowed: [go-marathon explicitly clears the former when setting the latter](https://github.com/gambol99/go-marathon/blob/15ea23e360abb8b25071e677aed344f31838e403/application.go#L152-L157) because the Marathon API would otherwise reject our API request. In that sense, it makes even less sense to set up both in the test (though I wouldn't relax the production code logic because of that because you never know how the Marathon API may evolve over time).